src/convertToObject.js

```diff
@@ -3,16 +3,32 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in [stylesString.js]
+ * (./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties (see an exampl in
+ * [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString.split(';')
+    .map(property => property.trim().split(':'))
+    .map(property => property.map(part => part.trim()))
+    .filter(property => property.length > 1)
+    .reduce(callback, {});
+}
+
+function callback(accumulator, property) {
+  const styleName = property[0];
+  const styleValue = property[1];
```

Comment:
More like advise, so u can use destructuring assignment here to avoid creating variables, 

`function callback(accumulator, [styleName, styleValue])`

Reply:
you can, but for this code `callback(accumulator, [styleName, styleValue])` more complicated than this one `callback(accumulator, property)`
Destructuring you can use in this way
```
function callback(accumulator, property) {
  const [styleName, styleValue] = property;
```